{Resource} Add example to tell how to pass array to --parameters (#13510)
Azure/azure-cli: doc/use_cli_effectively.md

    az vm list -d -g my_rg --query "[?powerState=='VM stopped'].id" -o tsv | az vm start --ids @-
    ```

## Pass complex parameter value (JSON/array/dict/file)

> Some parameters are able to receive complex value like JSON, array or combination of them, even reading content from a file.

Reviewer: How about we add some extra examples here for JSON objects?

> Their format varies in different parameters. For array, both `--parameter '("value1", "value2")'` or `--parameter '["value1", "value2"]'` could be valid.

Reviewer: ```suggestion
Their format varies with parameters. For list/array, both `--parameter '("value1", "value2")'` or `--parameter '["value1", "value2"]'` could be valid.
```

> Most important, we should take care of the quoting issue if you are a across different platforms user. Some quoting rules apply in different shells, please see [Quoting issues](https://github.com/Azure/azure-cli/blob/dev/doc/use_cli_effectively.md#quoting-issues).

Reviewer: ```suggestion
Most importantly, some quoting rules apply in different shells. Please see [Quoting issues](#quoting-issues) for more information.
```

    az vm list -d -g my_rg --query "[?powerState=='VM stopped'].id" -o tsv | az vm start --ids @-
    ```

## Pass complex parameter value (JSON/array/dict/file)

Some parameters are able to receive complex value like JSON, array or combination of them, even reading content from a file.

Their format varies with parameters.

> For list/array, both `--parameter '("value1", "value2")'` or `--parameter '["value1", "value2"]'` could be valid.

Reviewer: Let's add a full example with command here.

        --body '{"web":{"redirectUris":["https://myapp.com"]}}'
```

> ## Quoting issues - object/array parameter value

Reviewer: In my mind, this phrase `object/array parameter value` is unclear to the user what it means. What is an "object parameter"? What is an "array parameter"? It depends on specific commands. 

Also, this section is about quoting, instead of `passing object/array parameter value` (if I understand the meaning correctly.)

The change should go to doc https://docs.microsoft.com/en-us/cli/azure/group/deployment?view=azure-cli-latest which is where customer raised https://github.com/Azure/azure-cli/issues/10731, instead of this section.

Reviewer: As discussed, let's add a separate section dedicated to describe list/JSON object passing and link to this section with:

> Some quoting rules apply in different shells, please see [Quoting issues](#quoting-issues).

Reviewer: Actually I am think this may be related to https://github.com/Azure/azure-cli/issues/13170. We can make `--tags` support JSON as well to make it much easier for customers.
